app/models/concerns/presenters/statemachine/feature_in_states.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module Presenters::Statemachine
+  # This module provides a method to determine if a feature can be enabled
+  # based on the current state of the instance.
+  #
+  # State names:
+  # Presenters::StandardPresenter.state_machines[:state].states.map(&:value)
+  # pending, passed, started, cancelled, failed, processed_1, processed_2,
+  # processed_3, processed_4, qc_complete, unknown
+  module FeatureInStates
+    extend ActiveSupport::Concern
+
+    included do
+      # Determines if a feature can be enabled based on provided states.
+      #
+      # @param states [String, Symbol, Array, Hash] The states to check against.
+      # This can be a single state as a String or Symbol, an Array of states,
+      # or a Hash with includes and excludes keys for states.
+      # @param default [Boolean] The default return value if no specific
+      # condition is met.
+      # @return [Boolean] Returns true if the feature can be enabled based on
+      # the states parameter, otherwise returns the default.
+      #
+      # :reek:BooleanParameter
+      # :reek:ManualDispatch
+      # :reek:TooManyStatements
+      def can_be_enabled?(states, default: true)
+        return default unless respond_to?(:state) # No state method.
+        return default if states.blank? # No states to check.
+
+        current_state = state.to_s
+        return default if current_state.blank? # No current state.
+
+        includes, excludes = collect_includes_and_excludes(states)
+
+        # Exludes take precedence over includes.
```

Comment:
```suggestion
        # Excludes take precedence over includes.
```

Reply:
The binning algorithm is the same. What I rewrote is how the vac tubes are tracked. Previously it was using ancestor_tubes relation, now it is using sample_metadata supplier_name as you suggested.